### PR TITLE
Get Qemu to fingerprint and test properly on both windows and linux

### PR DESCRIPTION
**Pull request by dadgar**
On Windows `qemu-system-x86_64` does not respond to `--version` but `qemu-img` does. Also removed the precondition that you have to be root because you don't

### CODE REVIEW

File: client/testutil/driver_compatible.go [hunk 1]
```diff
@@ -14,10 +14,7 @@ func ExecCompatible(t *testing.T) {
 }
 
 func QemuCompatible(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		t.Skip("Must be on non-windows environments to run test")
-	}
-	// else see if qemu exists
+	// Check if qemu exists
 	_, err := exec.Command("qemu-system-x86_64", "-version").CombinedOutput()
```
Comment (achanda) [inline client/testutil/driver_compatible.go]: We need the same check here to run tests on windows

Comment (achanda) [inline client/testutil/driver_compatible.go]: We need the same check here to run tests on windows

Comment (dadgar) [inline client/testutil/driver_compatible.go]: It still works because that binary returns without an exit code but it is confusing. Updated

**Comment by diptanu** [comment]
@dadgar LGTM

File: client/driver/qemu.go [hunk 2]
```diff
@@ -56,13 +55,11 @@ func NewQemuDriver(ctx *DriverContext) Driver {
 }
 
 func (d *QemuDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
-	// Only enable if we are root when running on non-windows systems.
-	if runtime.GOOS != "windows" && syscall.Geteuid() != 0 {
-		d.logger.Printf("[DEBUG] driver.qemu: must run as root user, disabling")
-		return false, nil
+	bin := "qemu-system-x86_64"
```
Comment (diptanu) [inline client/driver/qemu.go]: Add a small comment here around why we are switching commands?

Comment (dadgar) [inline client/driver/qemu.go]: Done

**Pull request merged**